firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigGetParameterHandler.java

```diff
@@ -325,6 +337,33 @@ public FirebaseRemoteConfigValue getValue(String key) {
     return allConfigs;
   }
 
+  /**
+   * Adds a listener that will be called whenever one of the get methods is called.
+   *
+   * @param listener function that takes in the parameter key and the {@link ConfigContainer} JSON
+   */
+  public void addListener(BiConsumer<String, JSONObject> listener) {
+    synchronized (listeners) {
+      listeners.add(listener);
+    }
+  }
+
+  /**
+   * Calls all listeners in {@link #listeners}.
+   *
+   * @param key parameter key that was retrieved
+   * @param container the container the key was retrieved from
+   */
+  private void callListeners(String key, ConfigContainer container) {
+    JSONObject configs = container.getContainer();
+
+    synchronized (listeners) {
+      for (BiConsumer<String, JSONObject> listener : listeners) {
+        executor.execute(() -> listener.accept(key, configs));
```

Comment:
Thinking: I like this enforcement of the listener being called on a separate thread, since this is running in the low-latency `get` layer